components/gpio_control/gpio_control.py

```diff
@@ -6,115 +6,124 @@
 
-GPIO.setmode(GPIO.BCM)
-
-logger = logging.getLogger(__name__)
-
-
-def getFunctionCall(function_name):
-    try:
-        if function_name != 'None':
-            return getattr(function_calls, function_name)
-    except AttributeError:
-        logger.error('Could not find FunctionCall {function_name}'.format(function_name=function_name))
-    return lambda *args: None
-
-
-def generate_device(config, deviceName):
-    print(deviceName)
-    device_type = config.get('Type')
-    if deviceName.lower() == 'VolumeControl'.lower():
-        return VolumeControl(config)
-    elif device_type == 'TwoButtonControl':
-        logger.info('adding TwoButtonControl')
-        return TwoButtonControl(
-            config.getint('Pin1'),
-            config.getint('Pin2'),
-            getFunctionCall(config.get('functionCall1')),
-            getFunctionCall(config.get('functionCall2')),
-            functionCallTwoBtns=getFunctionCall(config.get('functionCallTwoButtons')),
-            pull_up=config.getboolean('pull_up', fallback=True),
-            hold_repeat=config.getboolean('hold_repeat', False),
-            hold_time=config.getfloat('hold_time', fallback=0.3),
-            name=deviceName)
-    elif device_type in ('Button', 'SimpleButton'):
-        return SimpleButton(config.getint('Pin'),
-                            action=getFunctionCall(config.get('functionCall')),
-                            name=deviceName,
-                            bouncetime=config.getint('bouncetime', fallback=500),
-                            edge=config.get('edge', fallback='FALLING'),
-                            hold_repeat=config.getboolean('hold_repeat', False),
-                            hold_time=config.getfloat('hold_time', fallback=0.3),
-                            pull_up_down=config.get('pull_up_down', fallback=GPIO.PUD_UP))
-    elif device_type == 'LED':
-        return LED(config.getint('Pin'),
-                            name=deviceName,
-                            initial_value=config.getboolean('initial_value', fallback=True))
-    elif device_type == 'MPDStatusLED':
-        return MPDStatusLED(config.getint('Pin'),
-                            host=config.get('host', fallback='localhost'),
-                            port=config.getint('port', fallback=6600),
-                            name=deviceName
-                            )
-    elif device_type == 'RotaryEncoder':
-        return RotaryEncoder(config.getint('pinUp'),
-                config.getint('pinDown'),
-                getFunctionCall(config.get('functionCallUp')),
-                getFunctionCall(config.get('functionCallDown')),
-                config.getfloat('timeBase', fallback=0.1),
+class gpio_control():
+
+    def __init__(self,function_calls):
+        self.devices = []
+        self.function_calls = function_calls
+        
+        GPIO.setmode(GPIO.BCM)
+        
+        lf = '%(asctime)s %(message)s'
+        lp = '/home/pi/RPi-Jukebox-RFID/logs/gpio.log'
```

Comment:
IIRC the old implementation was logging to syslog, which nicely places all logs into one place. I would like to avoid creating many different log files.

Reply:
yes, this is a mistake, thanks for point this out